server/manager/src/main/java/org/apache/accumulo/manager/compaction/coordinator/QueueMetrics.java

```diff
@@ -139,7 +140,16 @@ public void update() {
 
     SetView<CompactorGroupId> metricsWithoutQueues =
         Sets.difference(queuesWithMetrics, definedQueues);
-    metricsWithoutQueues.forEach(q -> {
+    Set<CompactorGroupId> toRemove = new HashSet<>(metricsWithoutQueues.size());
```

Comment:
There is another loop above this that is modifying perQeueMetrics while iterating over a set view that is based on it.  Maybe doing that following at the beginning of the method would avoid the problem for both loops.

```java
Set<CompactorGroupId> queuesWithMetrics = Set.copyOf(perQueueMetrics.keySet());
```